packages/base-shell/src/providers/Load/Provider.js

```diff
@@ -0,0 +1,65 @@
+import PropTypes from 'prop-types'
+import Context from './Context'
+import React, { useCallback, useReducer } from 'react'
+
+function reducer(response, action) {
+  switch (action.type) {
+    case 'LOADING':
+      return { ...initialState, displayText: action.payload.displayText || 'LOADING', statusLoad: 'loading', isLoading: true, isCanceled: false };
+    case 'LOADED':
+      return { ...initialState, displayText: action.payload.displayText || 'LOADED', ...action.payload, statusLoad: 'loaded', isLoading: false, ok: true, isCanceled: false };
+    case 'LOAD_CANCELED':
+      return { ...initialState, displayText: action.payload.displayText || 'LOAD CANCELED ', ...action.payload, statusLoad: 'load canceled', isLoading: false, ok: null, isCanceled: true };
+    case 'LOADING_WITH_ERROR':
+      return { ...initialState, displayText: action.payload.displayText || 'LOADING WITH ERROR ', ...action.payload, statusLoad: 'loading with error', isLoading: true, ok: false, isCanceled: false };
+    case 'LOADED_ERROR':
+      return { ...initialState, displayText: action.payload.displayText || 'LOADED ERROR ', ...action.payload, statusstatusLoadProcess: 'loaded error', isLoading: false, ok: false, isCanceled: false };
+    default:
+      return response;
+  }
+}
+
+const initialState = {
+  statusLoad: 'idle',
+  displayText: '',
+  isLoading: null,
+  isCanceled: null,
+  ok: null
+};
+
+function sleep() {
+  return new Promise((resolve) => {
+    setTimeout(resolve, 2000);
+  });
+}
+
+const Provider = ({ children }) => {
+  const [state, _dispatch] = useReducer(reducer, { ...initialState });
+
+  const dispatch = {
+    LOADING: useCallback((displayText) => {
+      _dispatch({ type: 'LOADING', payload: { displayText } });
+    }, [_dispatch]),
+    LOADED:  useCallback((payload) => {
+      _dispatch({ type: 'LOADED', payload: payload });
+    }, [_dispatch]),
+    LOAD_CANCELED:  useCallback((payload) => {
+      _dispatch({ type: 'LOAD_CANCELED', payload: payload });
+    }, [_dispatch]),
+    LOADED_ERROR:  useCallback((payload) => {
```

Comment:
loading with an error, 2 seconds to show a red preloader, to display an error loading